src/code/z_play.c

```diff
@@ -110,34 +110,254 @@
+        return SCENE_10YUKIYAMANOMURA;
+    }
+
+    // Spring Goron Village         -> Winter Goron Village
+    if (sceneNum == SCENE_11GORONNOSATO2) {
+        return SCENE_11GORONNOSATO;
+    }
+
+    // Spring Path to Goron Village -> Winter Path to Goron Village
+    if (sceneNum == SCENE_17SETUGEN2) {
+        return SCENE_17SETUGEN;
+    }
+
+    // Inverted Stone Tower         -> Stone Tower
+    if (sceneNum == SCENE_F41) {
+        return SCENE_F40;
+    }
+
+    return sceneNum;
+}
+
+/**
+ * Copies the flags set in ActorContext over to the current scene's CycleSceneFlags, usually using the original scene
+ * number. Exception for Inverted Stone Tower Temple, which uses its own.
+ */
+void Play_SaveCycleSceneFlags(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    CycleSceneFlags* cycleSceneFlags;
+
+    cycleSceneFlags = &gSaveContext.cycleSceneFlags[Play_GetOriginalSceneNumber(globalCtx->sceneNum)];
+    cycleSceneFlags->chest = globalCtx->actorCtx.flags.chest;
+    cycleSceneFlags->swch0 = globalCtx->actorCtx.flags.switches[0];
+    cycleSceneFlags->swch1 = globalCtx->actorCtx.flags.switches[1];
+
+    if (globalCtx->sceneNum == SCENE_INISIE_R) { // Inverted Stone Tower Temple
+        cycleSceneFlags = &gSaveContext.cycleSceneFlags[globalCtx->sceneNum];
+    }
+
+    cycleSceneFlags->collectible = globalCtx->actorCtx.flags.collectible[0];
+    cycleSceneFlags->clearedRoom = globalCtx->actorCtx.flags.clearedRoom;
+}
+
+void Play_SetRespawnData(GameState* gameState, s32 respawnMode, u16 entranceIndex, s32 roomIndex, s32 playerParams,
+                         Vec3f* pos, s16 yaw) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    gSaveContext.respawn[respawnMode].entranceIndex = Entrance_CreateIndex(entranceIndex >> 9, 0, entranceIndex & 0xF);
+    gSaveContext.respawn[respawnMode].roomIndex = roomIndex;
+    gSaveContext.respawn[respawnMode].pos = *pos;
+    gSaveContext.respawn[respawnMode].yaw = yaw;
+    gSaveContext.respawn[respawnMode].playerParams = playerParams;
+    gSaveContext.respawn[respawnMode].tempSwchFlags = globalCtx->actorCtx.flags.switches[2];
+    gSaveContext.respawn[respawnMode].unk_18 = globalCtx->actorCtx.flags.collectible[1];
+    gSaveContext.respawn[respawnMode].tempCollectFlags = globalCtx->actorCtx.flags.collectible[2];
+}
+
+void Play_SetupRespawnPoint(GameState* gameState, s32 respawnMode, s32 playerParams) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (globalCtx->sceneNum != SCENE_KAKUSIANA) { // Grottos
+        Play_SetRespawnData(&globalCtx->state, respawnMode, (u16)((void)0, gSaveContext.entranceIndex),
+                            globalCtx->roomCtx.currRoom.num, playerParams, &player->actor.world.pos,
+                            player->actor.shape.rot.y);
+    }
+}
+
+// Override respawn data in Sakon's Hideout
+void func_80169ECC(GlobalContext* globalCtx) {
```

Comment:
```suggestion
/**
 * Override respawn data in Sakon's Hideout
 */
void func_80169ECC(GlobalContext* globalCtx) {
```
Unless I'm mistaken, I think this is still the style for single line comments on documentation (comments below that are just temporary decomp nots for later are fine as //)

Reply:
Yes, I'm using line comments for temporary stuff.